components/Observer.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <div class="observer" />
+</template>
+
+<script>
+export default {
+  props: {
+    options: {
+      type: Object,
+      default: null
+    }
+  },
+  data: () => ({
+    observer: null
+  }),
+  mounted() {
+    const options = this.options || {}
+    // Use array destructing on the entry argument i.e. entries[0]
```

Comment:
Destructuring?

Reply:
The intersection observer callback receives an array of entries.  We are only interested in the first one i.e entries[0].  We can easily get this out by destructuring the array to [entry].

This was something introduced in ES6:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Reply:
re. "Yes, I just meant that I thought it was a typo - destructing should be destructuring"

Doh.  It's hard to type with cold fingers. :-)